test/deploy.py

```diff
@@ -0,0 +1,486 @@
+        ]
+        cmd = [
+            'ceph', 'orch', 'apply', 'osd', '--all-available-devices'
+        ]
+        self.checkOutputOnCephadmShell(vm_name, cmd)
+
+        for attempt in range(0, checkCount):
+            status = json.loads(self.checkOutputOnCephadmShell(vm_name, status_cmd))
+            osd_count = status['osdmap']['num_osds']
+            if osd_count >= expectedOSDNum:
+                break
+            print("Attempt {}: OSD not up! Count {}".format(attempt, osd_count))
+            time.sleep(60) # Wait for a minute
+
+        status = json.loads(self.checkOutputOnCephadmShell(vm_name, status_cmd))
+        osd_count = status['osdmap']['num_osds']
+        if osd_count < expectedOSDNum:
+            raise EnvironmentError("OSDs not up Count {}".format(osd_count))
+        print("OSD Count {}".format(osd_count))
+
+    def patchCephRuleForSingleNode(self, vm_name:string) -> None:
+        '''Creates a suitable replication rule for single node deployments.'''
+        # Create a new rule with OSD as failure domain.
+        new_rule_name = 'new_replication_rule_osd'
+        cmd = [
+            'ceph', 'osd', 'crush', 'rule', 'create-replicated',
+            new_rule_name, 'default', 'osd'
+        ]
+        self.checkOutputOnCephadmShell(vm_name, cmd)
+
+        # Make new rule default for all pools.
+        cmd = [
+           'ceph','osd','pool','ls', '-f', 'json'
+        ]
+        pools = json.loads(self.checkOutputOnCephadmShell(vm_name, cmd))
+        for pool in pools:
+            cmd = [
+                'ceph', 'osd', 'pool', 'set', pool, 'crush_rule',
+                new_rule_name
+            ]
+            self.checkOutputOnCephadmShell(vm_name, cmd)
+
+        # Fetch and delete other replication rules
+        cmd = [
+            'ceph', 'osd', 'crush', 'rule', 'dump', '-f', 'json'
+        ]
+        rules = json.loads(self.checkOutputOnCephadmShell(vm_name, cmd))
+        for rule in list(rules):
+            if rule['rule_name'] == new_rule_name:
+                continue
+            cmd = [
+                'ceph', 'osd', 'crush', 'rule', 'rm', rule['rule_name'],
+            ]
+            self.checkOutputOnCephadmShell(vm_name, cmd)
+
+def printScriptHelp() -> None:
+    print("Script Help")
+
+if __name__ == "__main__":
+    vm_name = ""
+    volumes = []
+    # total arguments
+    num_args = len(sys.argv)
+    # Deploy Cephadm over LXD if no arguments passed.
+    if num_args == 1:
+        try:
+            runner = DeployRunner()
+            runner.createStoragePool()
+            volumes = runner.createStorageVolume()
+            runner.createVmProfile(volumes)
+            vm_name = runner.createVm()
+            runner.syncRepoToVm(vm_name)
+            runner.modifyUserPasswd(vm_name)
+            runner.installAptPackage(vm_name)
+            runner.prepareContainerImage(vm_name)
+            runner.bootstrapCephadm()
+            time.sleep(60) #Wait for a minute.
```

Comment:
It would be preferable if we could probe for completion instead using a fixed sleep time. Due to load or other factors timings could vary

Reply:
Yeah, although mon-mgr are fast to comeup, it may be slowed down by the overall load on the lxd host. have changed to polling at 30 sec intervals.